### ISSUE TITLE
Validation Set During Training

### ISSUE THREAD

**Issue by Deltadahl** [2U1/Qwen2-VL-Finetune#22]
Would it be possible to use a validation set during training so that I can see if my model overfits?

**Comment by 2U1** [comment]
Yes you could, but you need to fix a the code a little bit, in `data.py` and `parametrs.py`.

**Comment by xjohnxjohn** [comment]
I need it too!

**Comment by whycantfindaname** [comment]
@2U1 Hello, i have fixed my code mainly in data.py and param.py for validation set. However I have encountered a problem when running eval:
> KeyError: "The metric_for_best_model training argument is set to 'eval_loss', which is not found in the evaluation metrics. 
It is the same problem in [33420](https://github.com/huggingface/transformers/issues/33420), it is mainly because PeftModel hides base models input arguments and its forward func has no labels signature.
I have tried add label_names="labels" and fix the trainer code according to [33420](https://github.com/huggingface/transformers/issues/33420), but they also failed. Could you have a look at this?
